src/UglyToad.PdfPig/DocumentLayoutAnalysis/Distances.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Linq;
+using UglyToad.PdfPig.Geometry;
+
+namespace UglyToad.PdfPig.DocumentLayoutAnalysis
+{
+    /// <summary>
+    /// Contains helpful tools for distance measures.
+    /// </summary>
+    public static class Distances
+    {
+        /// <summary>
+        /// The Euclidean distance is the "ordinary" straight-line distance between two points.
+        /// </summary>
+        /// <param name="point1">The first point.</param>
+        /// <param name="point2">The second point.</param>
+        /// <returns></returns>
+        public static double Euclidean(PdfPoint point1, PdfPoint point2)
+        {
+            double dx = (double)(point1.X - point2.X);
+            double dy = (double)(point1.Y - point2.Y);
+            return Math.Sqrt(dx * dx + dy * dy);
+        }
+
+        /// <summary>
+        /// The weighted Euclidean distance.
+        /// </summary>
+        /// <param name="point1">The first point.</param>
+        /// <param name="point2">The second point.</param>
+        /// <param name="wX">The weight of the X coordinates. Default is 1.</param>
+        /// <param name="wY">The weight of the Y coordinates. Default is 1.</param>
+        /// <returns></returns>
+        public static double WghtdEuclidean(PdfPoint point1, PdfPoint point2, double wX = 1.0, double wY = 1.0)
+        {
+            double dx = (double)(point1.X - point2.X);
+            double dy = (double)(point1.Y - point2.Y);
+            return Math.Sqrt(wX * dx * dx + wY * dy * dy);
+        }
+
+        /// <summary>
+        /// The Manhattan distance between two points is the sum of the absolute differences of their Cartesian coordinates.
+        /// <para>Also known as rectilinear distance, L1 distance, L1 norm, snake distance, city block distance, taxicab metric.</para>
+        /// </summary>
+        /// <param name="point1">The first point.</param>
+        /// <param name="point2">The second point.</param>
+        /// <returns></returns>
+        public static double Manhattan(PdfPoint point1, PdfPoint point2)
+        {
+            return (double)(Math.Abs(point1.X - point2.X) + Math.Abs(point1.Y - point2.Y));
+        }
+
+        /// <summary>
+        /// Find the nearest point.
+        /// </summary>
+        /// <param name="pdfPoint">The reference point, for which to find the nearest neighbour.</param>
+        /// <param name="points">The list of neighbours candidates.</param>
+        /// <param name="measure">The distance measure to use.</param>
+        /// <param name="dist">The distance between reference point, and its nearest neighbour</param>
+        /// <returns></returns>
+        public static PdfPoint FindNearest(this PdfPoint pdfPoint, PdfPoint[] points,
```

Comment:
I think here it would be better to use `IReadOnlyList<PdfPoint>` over `PdfPoint[]` since it allows the consumer to use a `List<PdfPoint>` or some custom type while providing the guarantee we won't mutate their collection inside our method. There's a slight performance penalty to doing that (which I'm about to contradict myself on) but I think in this case the benefits to the publicly visible API are worth it.

Reply:
Actually having thought about this it's probably even more convenient to make the argument:

```suggestion
        public static PdfPoint FindNearest(this PdfPoint pdfPoint, IEnumerable<PdfPoint> points,
```

Which would make the suggested algorithm below use a `foreach` loop (throwing if there's not at least one item in the enumerable).